ouroboros-consensus/src/Ouroboros/Storage/ChainDB/API.hs

```diff
@@ -438,18 +436,8 @@ traverseIterator f it = it {
       iteratorNext = iteratorNext it >>= traverse f
     }
 
--- | Equality instance for iterators
---
--- This relies on the iterator IDs assigned by the database.
---
--- NOTE: Iterators created by /different instances of the DB/ may end up with
--- the same ID. This should not matter in practice since there should not /be/
--- more than one DB, but it should nonetheless be noted.
-instance Eq (Iterator m blk b) where
-  (==) = (==) `on` iteratorId
-
-newtype IteratorId = IteratorId Int
-  deriving stock (Show)
+newtype IteratorId = IteratorId Word
```

Comment:
Rename it to `IteratorKey` + add a comment (why do we need it?)

Same for `ReaderId` + say why it's not a newtype.

Export them for testing purposes only.